core/src/main/java/org/everit/json/schema/ArraySchema.java

```diff
@@ -229,9 +229,12 @@ public boolean definesProperty(String field) {
     private boolean tryPropertyDefinitionByMetaIndex(String nextToken, String remaining, boolean hasRemaining) {
         boolean isAll = "all".equals(nextToken);
         boolean isAny = "any".equals(nextToken);
-        if (!hasRemaining && (isAll || isAny)) {
-            return true;
+        if (!hasRemaining) {
+            if (isAll || isAny) {
+                return true;
+            }
```

Comment:
This doesn't look like a useful change, I prefer staying on the current implementation.